go-controller/hybrid-overlay/cmd/hybrid-overlay-node/hybrid-overlay-node.go

```diff
@@ -38,22 +38,33 @@ func main() {
 			Destination: &nodeName,
 		},
 		&cli.BoolFlag{
-			Name:        "windows-service",
-			Usage:       "Enables hybrid overlay to run as a Windows service. Ignored on Linux.",
-			Destination: &runAsWindowsService,
+			Name:  windowServiceArgName,
+			Usage: "Enables hybrid overlay to run as a Windows service. Ignored on Linux.",
 		}})
+
+	ctx := context.Background()
+
 	c.Action = func(c *cli.Context) error {
+		if err := initForOS(c, &ctx); err != nil {
+			klog.Exit(err)
+		}
+
 		if err := runHybridOverlay(c); err != nil {
 			panic(err.Error())
 		}
 		return nil
 	}
 
-	ctx := context.Background()
+	if err := c.RunContext(ctx, os.Args); err != nil {
+		klog.Exit(err)
+	}
+}
+
+func signalHandler(c *context.Context) {
```

Comment:
@mansikulkarni96 context.Context is an interface, which means it's already a pointer, so we don't need an additional * here to make it pointer-to-pointer.